force-app/main/default/classes/NKS_CallFlowController.cls

```diff
@@ -21,19 +25,46 @@ public without sharing class NKS_CallFlowController {
 
         if (getEnquiryLogObject() == null && !Test.isRunningTest()) {
             for (String puzSessionId : sessionIds) {
-                respList.add(new CallQueueResponse('', 'PERSON', 'TEST QUEUE')); //Person is fallback when there is no match
+                respList.add(new CallQueueResponse('', 'PERSON', 'TEST QUEUE', '')); //Person is fallback when there is no match
             }
         } else {
             List<Object> enqLogs = getEnquiryLogs(sessionIds);
+            Map<String, Object> enqLogsMap = new Map<String, Object>();
             Set<String> queueNames = new Set<String>();
             for (Object enqLog : enqLogs) {
                 Map<String, Object> enqMap = (Map<String, Object>) enqLog;
+                if (enqMap.containsKey('puzzel__SessionId__c'))
+                    enqLogsmap.put((String) enqMap.get('puzzel__SessionId__c'), enqLog);
                 if (enqMap.containsKey('puzzel__Queue__c'))
                     queueNames.add((String) enqMap.get('puzzel__Queue__c'));
             }
 
-            for (Puzzel_Queue_Mapping__mdt queueMapping : getQueueMappings(queueNames)) {
-                respList.add(new CallQueueResponse(queueMapping.Theme_Group_Code__c, queueMapping.Caller_Context__c, queueMapping.Puzzel_Queue_Name__c));
+            List<Puzzel_Queue_Mapping__mdt> queueContextMapping = getQueueMappings(queueNames);
```

Comment:
Kan vi ta denne tilbake i for-loopen istedenfor å lagre som en variabel, så sparer vi heap-size? 🤓

Reply:
smart